engine-nextgen/src/main/kotlin/graphql/nadel/validation/NadelSchemaValidationError.kt

```diff
@@ -151,6 +151,44 @@ sealed interface NadelSchemaValidationError {
         override val subject = overallField
     }
 
+    data class IncompatibleFieldInputType(
+            val parentType: NadelServiceSchemaElement,
+            val overallInputField: GraphQLInputObjectField,
+            val underlyingInputField: GraphQLInputObjectField,
+    ) : NadelSchemaValidationError {
+        val service: Service get() = parentType.service
+
+        override val message = run {
+            val s = service.name
+            val of = makeFieldCoordinates(parentType.overall.name, overallInputField.name)
+            val uf = makeFieldCoordinates(parentType.underlying.name, underlyingInputField.name)
+            val ot = GraphQLTypeUtil.simplePrint(overallInputField.type)
+            val ut = GraphQLTypeUtil.simplePrint(underlyingInputField.type)
+            "Overall field $of has input type $ot but underlying field $uf in service $s has input type $ut"
+        }
+
+        override val subject = overallInputField
+    }
+
+    data class IncompatibleArgumentInputType(
+            val parentType: NadelServiceSchemaElement,
+            val overallInputArg: GraphQLArgument,
+            val underlyingInputArg: GraphQLArgument,
+    ) : NadelSchemaValidationError {
+        val service: Service get() = parentType.service
+
+        override val message = run {
+            val s = service.name
+            val ofa = makeFieldCoordinates(parentType.overall.name, overallInputArg.name)
+            val ufa = makeFieldCoordinates(parentType.underlying.name, underlyingInputArg.name)
+            val ot = GraphQLTypeUtil.simplePrint(overallInputArg.type)
+            val ut = GraphQLTypeUtil.simplePrint(underlyingInputArg.type)
+            "Overall field argument $ofa has input type $ot but underlying field argument $ufa in service $s has input type $ut"
```

Comment:
Does this message make sense?

For the argument

```
type Foo {
  baz(arg: String)
}
```

Doesn't it print out

> Overall field argument Foo.arg has input type String

It should be

> Overall field Foo.baz has argument arg of input type String …